Chart broken on Kubernetes <1.16 due to lookup using apiextensions.k8s.io/v1 CRD

Hi.

When using the chart with a cluster in version 1.15, the helm install fails with this error message:
```
lookup_func.go:111: [ERROR] unable to retrieve resource list for: apiextensions.k8s.io/v1 , error: the server could not find the requested resource
lookup_func.go:85: [ERROR] unable to get apiresource from unstructured: apiextensions.k8s.io/v1, Kind=CustomResourceDefinition , error the server could not find the requested resource
Error: template: kong/templates/custom-resource-definitions.yaml:21:25: executing "kong/templates/custom-resource-definitions.yaml" at <lookup "apiextensions.k8s.io/v1" "CustomResourceDefinition" "" "kongplugins.configuration.konghq.com">: error calling lookup: unable to get apiresource from unstructured: apiextensions.k8s.io/v1, Kind=CustomResourceDefinition: the server could not find the requested resource
```
This is using Helm 3.3.4, the last version of Helm compatible with the cluster.

The cause of the error is here https://github.com/Kong/charts/blob/kong-2.0.0-rc.1/charts/kong/templates/custom-resource-definitions.yaml#L21, directly using the `apiextensions.k8s.io/v1` API without first checking its availability.

The chart advertises compatibility with Kube 1.12+ so it looks like a bug :).

--- a/charts/kong/CHANGELOG.md
+++ b/charts/kong/CHANGELOG.md
@@ -1,5 +1,13 @@
 # Changelog
 
+## 2.0.0-rc.2
+
+### Fixed
+
+* The legacy CRD lookup now functions on Kubernetes versions that do not
+  support `apiextensions.k8s.io/v1/CustomResourceDefinition` (versions prior to
+  1.16).
+
 ## 2.0.0-rc.1
 
 ### Breaking changes

--- a/charts/kong/Chart.yaml
+++ b/charts/kong/Chart.yaml
@@ -10,5 +10,5 @@ maintainers:
   email: traines@konghq.com
 name: kong
 sources:
-version: 2.0.0-rc.1
+version: 2.0.0-rc.2
 appVersion: 2.3

--- a/charts/kong/templates/custom-resource-definitions.yaml
+++ b/charts/kong/templates/custom-resource-definitions.yaml
@@ -22,6 +22,7 @@
   {{- if .Capabilities.APIVersions.Has "apiextensions.k8s.io/v1/CustomResourceDefinition" -}}
     {{- $kongPluginCRD = (lookup "apiextensions.k8s.io/v1" "CustomResourceDefinition" "" "kongplugins.configuration.konghq.com") -}}
   {{- else -}}
+    {{/* TODO: remove the v1beta1 path when we no longer support k8s <1.16 */}}
     {{- $kongPluginCRD = (lookup "apiextensions.k8s.io/v1beta1" "CustomResourceDefinition" "" "kongplugins.configuration.konghq.com") -}}
   {{- end -}}
   {{- if $kongPluginCRD -}}

--- a/charts/kong/templates/custom-resource-definitions.yaml
+++ b/charts/kong/templates/custom-resource-definitions.yaml
@@ -18,7 +18,12 @@
        so this clause pretends the default didn't change if you have an existing release that relied
        on it
   */}}
-  {{- $kongPluginCRD := (lookup "apiextensions.k8s.io/v1" "CustomResourceDefinition" "" "kongplugins.configuration.konghq.com") -}}
+  {{- $kongPluginCRD := false -}}
+  {{- if .Capabilities.APIVersions.Has "apiextensions.k8s.io/v1/CustomResourceDefinition" -}}
+    {{- $kongPluginCRD = (lookup "apiextensions.k8s.io/v1" "CustomResourceDefinition" "" "kongplugins.configuration.konghq.com") -}}
+  {{- else -}}
+    {{- $kongPluginCRD = (lookup "apiextensions.k8s.io/v1beta1" "CustomResourceDefinition" "" "kongplugins.configuration.konghq.com") -}}
+  {{- end -}}
   {{- if $kongPluginCRD -}}
     {{- if (hasKey $kongPluginCRD.metadata "annotations") -}}
       {{- if (eq .Release.Name (get $kongPluginCRD.metadata.annotations "meta.helm.sh/release-name")) -}}

